### PR TITLE
Added WAV support to Sound

### PR DESCRIPTION
**Pull request by themuflon**

### CODE REVIEW

File: loom/common/assets/wavloader.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#include <string.h>
```
Comment (bengarney) [inline loom/common/assets/wavloader.cpp]: Header needed - where did this code come from?

File: loom/common/assets/wavloader.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#ifndef __wavloader__
```
Comment (bengarney) [inline loom/common/assets/wavloader.h]: Copyright header needed.

File: loom/common/assets/wavloader.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+typedef struct
+{
+    uint16_t numChannels;
+    uint32_t samplesPerSecond;
+    uint16_t sampleSize;
+    int32_t sampleDataSize;
+} wav_info;
+
+bool load_wav(const uint8_t* inData,
```
Comment (bengarney) [inline loom/common/assets/wavloader.h]: Probably document this function?

File: loom/engine/sound/lsSound.cpp [hunk 4]
```diff
@@ -116,6 +116,11 @@ class OALBufferNote
         buffer = 0;
         refCounter = 1;
     }
+    
+    ~OALBufferNote()
+    {
+        delete [] asset;
```
Comment (bengarney) [inline loom/engine/sound/lsSound.cpp]: Oh good catch. But should be lmDelete or something not a raw delete?

**Comment by bengarney** [comment]
Nice improvement. Nitted!

File: loom/engine/sound/lsSound.cpp (hunk 4 above)
Comment (themuflon) [inline loom/engine/sound/lsSound.cpp]: No, looks like that memory is allocated by openAL

Comment (themuflon) [inline loom/engine/sound/lsSound.cpp]: Whoops, messed up the mebers. It's actually allocated by strdup, so it's fine by itself. But maybe we should replace strup and allocated memory with lmNew?

File: loom/common/assets/wavloader.cpp (hunk 1 above)
Comment (themuflon) [inline loom/common/assets/wavloader.cpp]: After a relativley qucik search, I could not find code similar on the internet. It was probably written "from scratch" to use Loom support functions.

**Comment by bengarney** [comment]
OK, great - landing!

**Pull request merged**